### DISCUSSION TITLE
Dolby Atmos E-AC3 JOC HW decoding on Android

### DISCUSSION THREAD

**Discussion by xXxWeAreBorgxXx** [mpv-player/mpv#14765] [Q&A]
Is there way to use hw decoder for the E-AC3 JOC audio format on Android? It looks like SW decoder is always used, and it only decodes backwards-compatible E-AC3 part. Since Dolby Atmos decoder is proprietary, ffmpeg doesn't support it's decoding. 
Some other android players like stock or MX Player support hw audio decoder and therefore support Dolby Atmos on supported devices.
Setting ad=mediacodec or ad=mediacodex-copy doesn't solve the issue. Here is the file in which decoding of the JOC part can be easily checked at the moment 0:42: 
https://drive.google.com/file/d/1HxV_Xsjv9fMQhf0YcYjTRC1iL8e3okUQ/view?usp=drivesdk

**Comment by Andarwinux** [comment]
ffmpeg doesn't support audio mediacodec, but there may be plans in progress, see [ML](https://lists.ffmpeg.org/pipermail/ffmpeg-devel/2024-August/332807.html).
